lib/pliny/middleware/canonical_log_line.rb

```diff
@@ -0,0 +1,121 @@
+module Pliny::Middleware
+  # Emits a "canonical log line", i.e. a single log line that contains as much
+  # relevant information about a request as possible and which makes for a
+  # single convenient reference point to understand all the vitals of any
+  # single request.
+  #
+  # This default implementation contains some useful data to get a project
+  # started, but it's usually recommended to vendor this middleware into your
+  # project and start adding some custom fields. Some examples of those might
+  # be:
```

Comment:
@brandur I know for sure that when this lands, I will want to add other fields to the canonical log line. For example, on the GitHub integration, I would want to include the repository.

I do understand your reservations about bringing this into the template right away. But as I understand it, moving it to the template after a while, will also require a migration in user code. I.e. the namespace of the middleware will change.

Do you think we could do this in a way where the default lives in Pliny-land, but extensions live in user-land? It probably either means to add a context of sorts to the request, or allowing subclassing of the base type which can than be loaded.